### PR TITLE
MM-18013 Allow configuration of SAML crypto hashing algorithms

### PR DESCRIPTION
**Pull request by sbishel**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18013

#### Related Pull Requests

- Has server changes
https://github.com/mattermost/mattermost-server/pull/12362
https://github.com/mattermost/enterprise/pull/498

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -2922,6 +2922,98 @@ const AdminDefinition = {
...
+                        key: 'SamlSettings.DigestAlgorithm',
+                        label: t('admin.saml.digestAlgorithmTitle'),
+                        label_default: 'Digest Algorithm',
+                        isDisabled: it.either(
+                            it.stateIsFalse('SamlSettings.Enable'),
+                            it.stateIsFalse('SamlSettings.SignRequest'),
+                        ),
+                        options: [
+                            {
+                                value: 'http://www.w3.org/2000/09/xmldsig#sha1',
```
Comment (jespino) [inline components/admin_console/admin_definition.jsx]: I think I would expect in the options value and display name simpler things, like `SHA1` or `SHA256`, and then, based on that options transform it to the underneath XML reference to the signature.

Comment (jespino) [reply components/admin_console/admin_definition.jsx]: This is applicable to all the new options added in all the different fields.

Comment (sbishel) [reply components/admin_console/admin_definition.jsx]: @jespino I disagree with this. (2/5) I really don't think "simpler" names help in this case. I actually think they lead to more ambiguity. By using the spec name, the administrator knows exactly which algorithm is being selected. 

However, the spec names are in the descriptions, so if we want to use the names in parenthesis as the display_name ie.(RSAwithSHA1). I will make that change. 

Regarding the value, I don't understand why you would want to transform the value. Why would the value be anything other than what you want written to the config file.  (Although I will make the values constants.)

cc: @wiersgallak

Comment (jespino) [reply components/admin_console/admin_definition.jsx]: I think the display name is better as something more descriptive than a uri to a resource that defines it (we can add that kind of definition in the documentation or in the help_text if is needed). But here I think we should relay in the opinion of @wiersgallak 

About something different between the config file and the value here, is not what I would expect, I would expect it to be equal, but I think for that people that use manual configuration (writing the config file or using the cli to set config fields, for example in HA environments using ansible or something similar) the need of use the whole URI is harder. For me, in this case, the algorithm for signature is `rsa-sha1` no `http://www.w3.org/2000/09/xmldsig#rsa-sha1` that is just leaking an implementation detail into the config file. Anyway my opinion here is just 3/5 so... I'm ok with keep going with your current approach, but I would like to have another extra opinion here, what do you think @enahum ?

**Pull request merged**